src/Sulu/Bundle/ContactBundle/Resources/public/js/components/contact-selection/main.js

```diff
@@ -0,0 +1,420 @@
+                        {
+                            title: this.sandbox.translate('contact.contacts.title'),
+                            data: templates.data(this.domIds.contactSearch, this.domIds.contactList)
+                        }
+                    ]
+                );
+            }
+
+            if (!!this.options.account) {
+                slide.tabs = slide.tabs.concat(
+                    [
+                        {
+                            title: this.sandbox.translate('contact.accounts.title'),
+                            data: templates.data(this.domIds.accountSearch, this.domIds.accountList)
+                        }
+                    ]
+                );
+            }
+
+            if (slide.tabs.length === 0) {
+                this.sandbox.logger.error('contact and account disabled');
+
+                return;
+            }
+
+            this.sandbox.dom.append(this.$el, $element);
+            this.sandbox.start([
+                {
+                    name: 'overlay@husky',
+                    options: {
+                        triggerEl: this.$addButton,
+                        cssClass: 'contact-content-overlay',
+                        el: $element,
+                        removeOnClose: false,
+                        container: this.$el,
+                        instanceName: 'contact-selection.' + this.options.instanceName + '.add',
+                        skin: 'wide',
+                        slides: [slide]
+                    }
+                }
+            ]);
+        },
+
+        getAddOverlayData = function() {
+            var accountDef = this.sandbox.data.deferred(),
+                contactDef = this.sandbox.data.deferred(),
+                data = [],
+                oldData = this.getData();
+
+            this.sandbox.emit('husky.datagrid.contact.items.get-selected', function(selected) {
+                this.sandbox.util.foreach(selected, function(item) {
+                    var value = 'c' + item,
+                        index = oldData.indexOf(value);
+
+                    if (index !== -1) {
+                        data[index] = value;
+                    } else {
+                        data.push(value);
+                    }
+                }.bind(this));
+
+                contactDef.resolve();
+            }.bind(this));
+            this.sandbox.emit('husky.datagrid.account.items.get-selected', function(selected) {
+                this.sandbox.util.foreach(selected, function(item) {
+                    var value = 'a' + item,
+                        index = oldData.indexOf(value);
+
+                    if (index !== -1) {
+                        data[index] = value;
+                    } else {
+                        data.push(value);
+                    }
+                }.bind(this));
+
+                accountDef.resolve();
+            }.bind(this));
+
+            this.sandbox.dom.when(accountDef, contactDef).then(function() {
```

Comment:
hmm you cant say it in general ... if the listener does some async magic then not (: here i think its synchronous but you never know if it changes anytime!